pkg/controller/machinedeployment/controller.go

```diff
@@ -134,12 +128,30 @@ func (r *ReconcileMachineDeployment) getMachineSetsForDeployment(d *v1alpha1.Mac
 			continue
 		}
 
+		// Attempt to adopt machine if it meets previous conditions and it has no controller references.
+		if metav1.GetControllerOf(ms) == nil {
+			if err := r.adoptOrphan(d, ms); err != nil {
+				klog.Warningf("Failed to adopt MachineSet %q into MachineDeployment %q: %v", ms.Name, d.Name, err)
+				continue
+			}
+		}
+
+		if !metav1.IsControlledBy(ms, d) {
+			continue
+		}
+
 		filteredMS = append(filteredMS, ms)
 	}
 
 	return filteredMS, nil
 }
 
+func (r *ReconcileMachineDeployment) adoptOrphan(d *v1alpha1.MachineDeployment, ms *v1alpha1.MachineSet) error {
```

Comment:
`d` is never referenced in this function

Reply:
Nice catch!